internal/checkers/http_const.go

```diff
@@ -0,0 +1,122 @@
+package checkers
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// HTTPConst detects situations like
+//
+//	assert.HTTPStatusCode(t, handler, "GET", "/index", nil, 200)
+//	assert.HTTPBodyContains(t, handler, "GET", "/index", nil, "counter")
+//
+// and requires
+//
+//	assert.HTTPStatusCode(t, handler, http.MethodGet, "/index", nil, http.StatusOK)
+//	assert.HTTPBodyContains(t, handler, http.MethodGet, "/index", nil, "counter")
+type HTTPConst struct{}
+
+// NewHTTPConst constructs Regexp checker.
+func NewHTTPConst() HTTPConst  { return HTTPConst{} }
+func (HTTPConst) Name() string { return "http-const" }
+
+func (checker HTTPConst) Check(_ *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
+	switch call.Fn.NameFTrimmed {
+	case "HTTPBody",
+		"HTTPBodyContains",
+		"HTTPBodyContainsf",
+		"HTTPBodyNotContains",
+		"HTTPBodyNotContainsf",
+		"HTTPError",
+		"HTTPErrorf",
+		"HTTPRedirect",
+		"HTTPSuccess",
+		"HTTPSuccessf":
+
+		if len(call.Args) < 2 {
+			return nil
+		}
+
+		if bt, ok := safeTypedBasicLit(call.Args[1], token.STRING); ok {
+			currentVal := unquoteBasicLitValue(bt)
+			key := strings.ToUpper(currentVal)
+			if newVal, ok := httpMethod[key]; ok {
+				return newHTTPConstDiagnostic(checker.Name(), call, []analysis.SuggestedFix{newConstReplacement(bt, currentVal, newVal)})
+			}
+			return nil
+		}
+	case "HTTPStatusCode", "HTTPStatusCodef":
+		if len(call.Args) < 5 {
+			return nil
+		}
+		var suggestedFixes []analysis.SuggestedFix
+		if bt, ok := safeTypedBasicLit(call.Args[1], token.STRING); ok {
+			currentVal := unquoteBasicLitValue(bt)
+			key := strings.ToUpper(currentVal)
+			if newVal, ok := httpMethod[key]; ok {
+				suggestedFixes = append(suggestedFixes, newConstReplacement(bt, currentVal, newVal))
+			}
+		}
+		if bt, ok := safeTypedBasicLit(call.Args[4], token.INT); ok {
+			currentVal := bt.Value
+			key := strings.ToUpper(currentVal)
```

Comment:
I was looking for this, please add tests for for lower case "get" or Pascal case (Get) and all other methods